package omnivoxel.client.game.graphics.api.opengl.mesh.meshData;

import omnivoxel.client.game.entity.EntityMeshWrapper;
import org.joml.Matrix4f;

import java.util.List;

public interface EntityMeshData extends MeshData {
    Matrix4f getModel();

    EntityMeshData setModel(Matrix4f model);

    void addChild(EntityMeshData entityMeshData);

    EntityMeshWrapper entity();

    List<EntityMeshData> children();
}